packages/react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -7,35 +7,73 @@ import { CheckboxRadioTemplate } from '../_CheckboxRadioTemplate';
 import classes from './Checkbox.module.css';
 
 export interface CheckboxProps {
+  /**
+   * An ID that will be applied to the ID attribute of the checkbox element.
+   * It will be automatically generated to a random value if not set.
+   */
   checkboxId?: string;
+
+  /**
+   * Set to `true` to disable the checkbox.
+   * This is currently the only way to set the `checked` mode, as uncontrolled mode is not supported yet.
+   */
   checked?: boolean;
+
+  /** Set to `true` to apply compact styling, making the checkbox smaller. */
   compact?: boolean;
+
+  /** The description of the checkbox. This will appear below the label. */
   description?: ReactNode;
+
+  /** Set to `true` to disable the checkbox. */
   disabled?: boolean;
+
+  /** Set to `true` to apply error styling to the checkbox. */
   error?: boolean;
+
+  /** Set this to make a clickable help icon appear to the right for the label. The content will appear when the user clicks on it. */
   helpText?: ReactNode;
+
+  /** Set to `true` to visually hide the label text, but still make it accessible to tools like screen readers. */
   hideLabel?: boolean;
+
+  /** The label of the checkbox. */
   label?: ReactNode;
+
+  /** The name of the checkbox. */
   name?: string;
+
+  /**
+   * A callback that will be called when the checkbox is toggled.
+   * As uncontrolled mode is not supported yet, this function has to be used to toggle between checked and unchecked state.
```

Comment:
I was abit confused by this wording. You are not using this function to toggle between checked or unchecked are you? `onChange` handlers normally just gives you the current checked value, so something like that?

Reply:
I'll see if I can formulate it better 🙂